File: leetcode/rust-leetcode/src/35.search-insert-position.rs
impl Solution {
    pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
        match nums.binary_search(&target) {
            Ok(i) => i as i32,
            Err(i) => i as i32
        }
    }
}

/* */

impl Solution {
    pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
        let (mut l, mut r) = (0, nums.len());
        while l < r {
            let mid = l + ((r - l) >> 1);
            if nums[mid] < target {
                l = mid + 1;
            } else {
                r = mid;
            }
        }
        l as _
    }
}

File: leetcode/go/35.search-insert-position.go
func searchInsert(nums []int, target int) int {
    l, r, mid := 0, len(nums), 0
    for l < r {
        mid = (l + r)/2
        if nums[mid] == target {
            return mid
        }
        if nums[mid] < target {
            l = mid + 1
        } else {
            r = mid
        }
    }
    return l
}

